trin-core/src/jsonrpc/utils.rs

```diff
@@ -0,0 +1,52 @@
+use crate::portalnet::Enr;
+use discv5::{
+    enr::NodeId,
+    kbucket::{ConnectionState, NodeStatus},
+};
+use ethereum_types::U256;
+use serde_json::{json, Value};
+use std::collections::HashMap;
+
+type NodeMap = HashMap<String, String>;
+type NodeTuple = (NodeId, Enr, NodeStatus, U256);
+
+/// Converts the output of the Overlay's bucket_entries method to a JSON Value
+pub fn bucket_entries_to_json(bucket_entries: HashMap<usize, Vec<NodeTuple>>) -> Value {
+    let mut node_count: u16 = 0;
+    let mut connected_count: u16 = 0;
+    let buckets_indexed: HashMap<usize, Vec<NodeMap>> = bucket_entries
```

Comment:
When the buckets are collected into a `HashMap` here, I think we also risk losing the ordering. Maybe it's sufficient to do a `Vec<(usize, Vec<NodeMap>)>`?

If that doesn't work, it appears that a [BTreeMap](https://doc.rust-lang.org/std/collections/struct.BTreeMap.html) might do the trick.

Reply:
iirc, I was originally doing that `Vec` based suggestion and `clippy` yelled at me for trying to use "too complex of a type".